website/content/docs/connect/proxies/envoy-extensions/usage/lua.mdx

```diff
@@ -0,0 +1,185 @@
+
+<Tabs>
+<Tab heading="HCL" group="hcl">
+<CodeBlockConfig filename="lua-envoy-extension-proxy-defaults.hcl">
+
+```hcl
+Kind = "proxy-defaults"
+Name = "global"
+Protocol  = "http"
+EnvoyExtensions {
+  Name = "builtin/lua"
+  
+  Arguments = {
+    ProxyType = "connect-proxy"
+    Listener = "inbound"
+    Script = <<-EOS
+function envoy_on_request(request_handle)
+  meta = request_handle:streamInfo():dynamicMetadata()
+  m = meta:get("consul")
+  request_handle:headers():add("x-consul-service", m["service"])
+  request_handle:headers():add("x-consul-namespace", m["namespace"])
+  request_handle:headers():add("x-consul-datacenter", m["datacenter"])
+  request_handle:headers():add("x-consul-trust-domain", m["trust-domain"])
+end
+EOS
+  }
+}
+```
+
+</CodeBlockConfig>
+</Tab>
+<Tab heading="JSON" group="json">
+<CodeBlockConfig filename="lua-envoy-extension-proxy-defaults.json">
+
+```hcl
+{
+  "kind": "proxy-defaults",
+  "name": "global",
+  "protocol": "http",
+  "envoy_extensions": [{
+    "name": "builtin/lua",
+    "arguments": {
+      "proxy_type": "connect-proxy",
+      "listener": "inbound",
+      "script": "function envoy_on_request(request_handle)\nmeta = request_handle:streamInfo():dynamicMetadata()\nm = \nmeta:get("consul")\nrequest_handle:headers():add("x-consul-service", m["service"])\nrequest_handle:headers():add("x-consul-namespace", m["namespace"])\nrequest_handle:headers():add("x-consul-datacenter", m["datacenter"])\nrequest_handle:headers():add("x-consul-trust-domain", m["trust-domain"])\nend"
+    }
+  }]
+}
+```
+
+</CodeBlockConfig>
+</Tab>
+<Tab heading="Kubernetes" group="kubernetes">
+<CodeBlockConfig filename="lua-envoy-extension-proxy-defaults.yaml">
+
+```yaml
+apiVersion: consul.hashicorp.com/v1alpha1
+kind: ProxyDefaults
+metadata:
+  name: global
+spec:
+  protocol: http
+  envoyExtensions:
+    name = "builtin/lua"
+    arguments:
+      proxyType: "connect-proxy"
+      listener: "inbound"
+      script: |-
+function envoy_on_request(request_handle)
+  meta = request_handle:streamInfo():dynamicMetadata()
+  m = meta:get("consul")
+  request_handle:headers():add("x-consul-service", m["service"])
+  request_handle:headers():add("x-consul-namespace", m["namespace"])
+  request_handle:headers():add("x-consul-datacenter", m["datacenter"])
+  request_handle:headers():add("x-consul-trust-domain", m["trust-domain"])
+end
+```
+
+</CodeBlockConfig>
+</Tab>
+</Tabs>
+
+For a full list of parameters for `EnvoyExtensions`, refer to the [`service-defaults`](/consul/docs/connect/config-entries/service-defaults#envoyextensions) and [`proxy-defaults`](/consul/docs/connect/config-entries/proxy-defaults#envoyextensions) configuration entries reference documentation.
+
+~> **Note:** Although you can apply `EnvoyExtensions` to `ProxyDefaults`, this may have unintended consequences. We recommend enabling `EnvoyExtensions` with `ServiceDefaults` in most cases.
```

Comment:
```suggestion
!>**Compatibility warning**: Applying `EnvoyExtensions` to `ProxyDefaults` may produce unintended consequences. We recommend enabling `EnvoyExtensions` with `ServiceDefaults` in most cases.
```
I don't love it, but I think this is probably the most applicable callout. I'd prefer that we just use a regular paragraph.

Reply:
I'm not sure if this is a compatibility issue...